packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/categories/UpdateCategoriesTests.java

```diff
@@ -0,0 +1,235 @@
+/*
+ * #%L
+ * Alfresco Remote API
+ * %%
+ * Copyright (C) 2005 - 2022 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+
+package org.alfresco.rest.categories;
+
+import static java.util.UUID.randomUUID;
+
+import static org.alfresco.utility.data.RandomData.getRandomName;
+import static org.alfresco.utility.report.log.Step.STEP;
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+import static org.springframework.http.HttpStatus.FORBIDDEN;
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+import static org.springframework.http.HttpStatus.OK;
+
+import org.alfresco.rest.model.RestCategoryModel;
+import org.alfresco.utility.model.FolderModel;
+import org.alfresco.utility.model.SiteModel;
+import org.alfresco.utility.model.TestGroup;
+import org.alfresco.utility.model.UserModel;
+import org.testng.annotations.Test;
+
+public class UpdateCategoriesTests extends CategoriesRestTest
+{
+    private static final String IGNORE_FIELD_NAME = FIELD_NAME;
+
+    /**
+     * Update a category (direct child of root category)
+     */
+    @Test(groups = { TestGroup.REST_API})
+    public void testUpdateCategory_asAdmin()
+    {
+        STEP("Prepare as admin a category under root category");
+        final RestCategoryModel createdCategory = prepareCategoryUnderRoot();
+
+        STEP("Update as admin newly created category");
+        final String newName = getRandomName("NewCategoryName");
+        final RestCategoryModel fixedCategoryModel = createCategoryModelOf(newName, createdCategory);
+        final RestCategoryModel updatedCategory = restClient.authenticateUser(dataUser.getAdminUser())
+            .withCoreAPI()
+            .usingCategory(createdCategory)
+            .updateCategory(fixedCategoryModel);
+
+        restClient.assertStatusCodeIs(OK);
+        updatedCategory.assertThat().isEqualTo(createdCategory, IGNORE_FIELD_NAME);
+        updatedCategory.assertThat().field(FIELD_NAME).isNot(createdCategory.getName());
+        updatedCategory.assertThat().field(FIELD_NAME).is(newName);
```

Comment:
Slightly redundant checks here, but maybe it makes the messages nicer if the obvious thing goes wrong.

Reply:
I think the same and would leave it as is.